drivers/usb/udc/udc_stm32.c

```diff
@@ -353,6 +353,16 @@ void HAL_PCD_DataInStageCallback(PCD_HandleTypeDef *hpcd, uint8_t epnum)
 	}
 }
 
+#if DT_INST_NODE_HAS_PROP(0, disconnect_gpios)
```

Comment:
It was building but nothing was happening because `HAL_PCDEx_SetConnectionState` is defined as a weak no-op in the hal, so on a device that needs the pullup to be controlled by a gpio the device would just not enumerate, this is fairly common on stm32f1 boards